biogeochem/EDPhysiologyMod.F90

```diff
@@ -503,42 +553,111 @@ subroutine trim_canopy( currentSite )
                    currentCohort%leaf_cost = currentCohort%leaf_cost * &
                          (EDPftvarcon_inst%grperc(ipft) + 1._r8)
                 endif
-                if (currentCohort%year_net_uptake(z) < currentCohort%leaf_cost)then
-                   if (currentCohort%canopy_trim > EDPftvarcon_inst%trim_limit(ipft))then
 
-                      if ( debug ) then
-                         write(fates_log(),*) 'trimming leaves', &
-                               currentCohort%canopy_trim,currentCohort%leaf_cost
-                      endif
+                ! Construct the arrays for a least square fit of the net_net_uptake versus the cumulative lai
+                if (currentCohort%nv > nll .and. currentCohort%nv - z < nll) then  ! Only for nll layers
+                  nnu_clai_a(1,1) = nnu_clai_a(1,1) + 1 ! Increment for each layer used
+                  nnu_clai_a(1,2) = nnu_clai_a(1,2) + currentCohort%year_net_uptake(z) - currentCohort%leaf_cost
+                  nnu_clai_a(2,1) = nnu_clai_a(1,2)
+                  nnu_clai_a(2,2) = nnu_clai_a(2,2) + (currentCohort%year_net_uptake(z) - currentCohort%leaf_cost)**2
+                  nnu_clai_b(1,1) = nnu_clai_b(1,1) + cumulative_lai
+                  nnu_clai_b(2,1) = nnu_clai_b(2,1) + (cumulative_lai * & 
+                                    (currentCohort%year_net_uptake(z) - currentCohort%leaf_cost))
+                end if
+
+                ! Check leaf cost against the yearly net uptake for that cohort leaf layer
+                if (currentCohort%year_net_uptake(z) < currentCohort%leaf_cost) then
```

Comment:
@glemieux , why do we allow wholesale trimming of the leaf layer in these conditions?  I thought the idea was to let the regression identify the intersection point and just go with that value. It looks like the regression would also over-write whatever was calculated here, so it seems unnecessary right?  Also, does the final solution adhere to the trim-limit?

Reply:
We do this in case we have less than `nll` leaf layers to work with for a given cohort.  In that case I just decided to let it stay with the old trimming methodology, baring some other rationale.  There are two thoughts I had on this:

1) I wasn't sure what we should do if we have a cohort with just one leaf layer come in to the routine `nv = 1`.  Presumably that would be for a new cohort so maybe we'd want to just leave it at the default starting trim fraction?
2) Right now `nll` is somewhat randomly hard coded to `4` for all cohorts regardless of number of leaf layer, `nv`.  I was thinking that there might be 'better' optimum intercepts if the `nll` layer is greater for cohorts with more leaf layers (up to a point).  That said, I could make the value of `nll` dependent on `nv` so that we'd never have to use the old method except for `nv = 1`.  Alternatively, I just set this to `nll = 2` and call it a day.